tests/test_modular_domains/results_true.dat

```diff
@@ -0,0 +1,5 @@
+# Iterations: 28
+keff:  8.58048E-01
```

Comment:
My only issue is a floating point artifact (keff = 8.5804**9**E-01 on my machine). Since that isn't unique to this PR, I shall approve.

Reply:
Well, this type of artifact can happen if a track is completely parallel to and advancing on a surface, then based on random fp math assigned on one side of the other. I'll examine this test further. If I don't find anything I suggest you merge and we'll deal with it if it spreads to other machines

Reply:
I think the reason might be CMFD boundary cut the cells into pieces, and some very tiny source regions are created.